spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java

```diff
@@ -389,23 +375,23 @@ public void close() {
 	}
 
 	@Override
-	public void delete(String lock) {
-		this.defaultTransactionTemplate.executeWithoutResult(
-				transactionStatus -> this.template.update(this.deleteQuery, this.region, lock, this.id));
+	public boolean delete(String lock) {
```

Comment:
This also presents an opportunity to check for `template.update() > 1` to throw exception, this will help prevent incorrect constraint on Table. 

This is possible only if table is created without `constraint INT_LOCK_PK primary key (LOCK_KEY, REGION)`.

Reply:
This is actually outside the scope of the PR. (my bad, I should not include this change). 
The method has been maintained in PR #9292. 
I think business logic should not address cases of misconfiguration or incorrect settings. 
If you'd like to discuss this method further, it would be great to create a new issue.